### DISCUSSION TITLE
Help required getting Solis S5-GR3P12K working

### DISCUSSION THREAD

**Discussion by suikerman** [wills106/homeassistant-solax-modbus#912] [Q&A]
### Describe the bug

Can anyone tell me if S5-GR3P12K s/n 180507 has been added, this would have been done with update 2024.03.3.

I have checked everything settings several times but I can't get it to work. Everything works fine in the cloud.
I have to use S2-wl-st
s/n 7A12320463B0...

I have no idea where to look for the problem.
can anyone help

### Integration Version

2024.05.1

### Homeassistant core version

2024.5.3

### Inverter brand

Solis

### Plugin used

plugin_solis.py

### Serial prefix

180507 

### Connection Method

RS485 - Lan

### Detailed Error Log

This error was caused by a custom integration.

Logger: custom_components.solax_modbus.plugin_solis
Source: custom_components/solax_modbus/plugin_solis.py:2457
integration: SolaX Inverter Modbus (documentation, problems)
First occurred: 17:27:30 (1 events)
Last logged: 5:27:30 PM

unrecognized solis inverter type - serial number : unknown
-------------------------------------------------- ------------------------------------------------
This error was caused by a custom integration.

Logger: custom_components.solax_modbus.plugin_solis
Source: custom_components/solax_modbus/plugin_solis.py:2431
integration: SolaX Inverter Modbus (documentation, problems)
First occurred: 17:27:30 (1 events)
Last logged: 5:27:30 PM

solis: cannot find serial number, even not for other Inverter
-------------------------------------------------- ------------------------------------------
This error was caused by a custom integration.

Logger: custom_components.solax_modbus.plugin_solis
Source: custom_components/solax_modbus/plugin_solis.py:70
integration: SolaX Inverter Modbus (documentation, problems)
First occurred: 17:27:30 (1 events)
Last logged: 5:27:30 PM

solis: reading serial number from address 0x80ec failed; other address may succeed
-------------------------------------------------- --------------------------------------------------
This error was caused by a custom integration.

Logger: custom_components.solax_modbus.plugin_solis
Source: custom_components/solax_modbus/plugin_solis.py:60
integration: SolaX Inverter Modbus (documentation, problems)
First occurred: 17:27:30 (1 events)
Last logged: 5:27:30 PM

solis: attempt to read serial number failed at 0x80ec
Traceback (most recent call last):
   File "/config/custom_components/solax_modbus/plugin_solis.py", line 60, in async_read_serialnr
     inverter_data = await hub.async_read_input_registers(unit=hub._modbus_addr, address=address, count=8)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   File "/config/custom_components/solax_modbus/__init__.py", line 491, in async_read_input_registers
     resp = await self._client.read_input_registers(address, count, **kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ^^^^^^^^^^^^^^^
   File "/usr/local/lib/python3.12/site-packages/pymodbus/client/base.py", line 183, in async_execute
     raiseModbusIOException(
pymodbus.exceptions.ModbusIOException: Modbus Error: [Input/Output] ERROR: No response received after 3 retries
-------------------------------------------------- ----------------------------
Logger: homeassistant.util.loop
Source: util/loop.py:99
First occurred: 17:27:10 (1 events)
Last logged: 17:27:10

Detected blocking call to import_module inside the event loop by custom integration 'solax_modbus' at custom_components/solax_modbus/__init__.py, line 143: plugin = importlib.import_module( (offender: /config/custom_components/solax_modbus/__init__.py, line 143: plugin = importlib.import_module(), please create a bug report at https://github.com/wills106/homsassistant-solax-modbus/issues

**Comment by stewwy** [comment]
Had the same error, in Vircon make sure the TCP connection shows Established between the 2 Waveshare devices before you start home assistant to configure Solax. Took me a week until I found the error, I found it best to reset both the waveshare devices to defaults a couple of times and check everything several times before it was established. Once it was established via DHCP I set everything to static IP's in the router to lock in everything.  I couldn't get more than 9600 on the server working, but that's plenty fast enough for HA
